Formula/noti.rb

```diff
@@ -17,19 +16,20 @@ class Noti < Formula
   depends_on "go" => :build
 
   def install
-    mkdir_p "#{buildpath}/src/github.com/variadico"
-    ln_s buildpath, "#{buildpath}/src/github.com/variadico/noti"
     ENV["GOPATH"] = buildpath
-    cd "#{buildpath}/src/github.com/variadico/noti" do
-      if build.stable?
-        system "go", "build", "-o", "#{bin}/noti"
-      else
-        system "go", "build", "-o", "#{bin}/noti", "cmd/noti/main.go", "cmd/noti/osx.go"
-      end
+
+    # Move files to right location in GOPATH.
+    notipath = buildpath/"src/github.com/variadico/noti"
+    notipath.install Dir["{*}"]
+
+    cd "src/github.com/variadico/noti/cmd/noti" do
+      system "go", "build"
+      bin.install "noti"
     end
   end
 
   test do
-    system "#{bin}/noti", "-t", "Noti", "-m", "'Noti recipe installation test has finished.'"
+    output = shell_output(bin/"noti -version")
```

Comment:
`noti` is a proxy for another command. So, `noti` always returns the exit code of the program it ran. Regardless of whether `noti` itself had any problems. If the proxied command returns `1`, then `noti` will return `1`.

`noti -t Noti -m 'Noti recipe installation test has finished.'` will always succeed because it's not running any other program. If the notification fails to trigger, an error will be written to stderr, but the exit status will be `0`.